substrate/runtime/staking/src/tests.rs

```diff
@@ -343,6 +341,28 @@ fn nominating_and_rewards_should_work() {
 	});
 }
 
+#[test]
+fn rewards_with_off_the_table_should_work() {
+	with_externalities(&mut new_test_ext(0, 1, 1, 0, true, 10), || {
+		System::set_block_number(1);
+		assert_ok!(Staking::stake(&1));
+		assert_ok!(Staking::nominate(&2, 1.into()));
+		assert_ok!(Staking::stake(&3));
+		Session::check_rotate_session();
+		assert_eq!(Session::validators(), vec![1, 3]);	// 1 + 2, 3
+		assert_eq!(Balances::total_balance(&1), 10);
+		assert_eq!(Balances::total_balance(&2), 20);
+		assert_eq!(Balances::total_balance(&3), 30);
+
+		System::set_block_number(2);
+		assert_ok!(Staking::register_preferences(&1, Staking::intentions().into_iter().position(|i| i == 1).unwrap() as u32, 3, 4));
+		Session::check_rotate_session();
```

Comment:
Oh wait, I'd expect that it should not build since https://github.com/paritytech/substrate/pull/646 introduced an argument here